src/network.js

```diff
@@ -33,29 +35,36 @@ class Network {
     this._onConnection = this._onConnection.bind(this)
   }
 
-  start () {
+  async start () {
     this._running = true
     this.libp2p.handle(this.protocols, this._onConnection)
 
-    this.libp2p.on('peer:connect', this._onPeerConnect)
-    this.libp2p.on('peer:disconnect', this._onPeerDisconnect)
+    // register protocol with topology
+    const topology = new MulticodecTopology({
+      multicodecs: this.protocols,
+      handlers: {
+        onConnect: this._onPeerConnect,
+        onDisconnect: this._onPeerDisconnect
+      }
+    })
+    this._registrarId = await this.libp2p.registrar.register(topology)
```

Comment:
Does this need to be async?  The `Registrar.register` method doesn't appear to be an async method: https://github.com/libp2p/js-libp2p/blob/master/src/registrar.js#L64-L78

Reply:
lol, yes my bad! 😓We had it async before, but not anymore. I will change this according

Reply:
just amended the previous commit removing the async part, as well as the breaking change message. Thanks for noticing this